templates/ui.html

```diff
@@ -1177,17 +1177,110 @@
               message: "",
               URL: "/user#/subscription?type=subscription",
             },
+            analyticsProductInfo: {
+              itemsMapping: {},
+              itemsInLists: {},
+            },
           };
         },
         methods: {
+          getItemsMapping(callback = null) {
+            fetch(
+              '/api/analytics/items',
+              {
+                method: 'GET',
+                credentials: 'include',
+              }
+            )
+            .then(response => {
+              if (response.status === 200) {
+                return response.json();
+              }
+              return Promise.reject(response);
+            })
+            .then((data) => {
+              this.analyticsProductInfo.itemsMapping = data.items_mapping;
+              this.analyticsProductInfo.itemsInLists = data.items_in_lists;
+            })
+            .catch((error) => {
+              console.error('loadItemMappingError error:', error);
+            })
+            .finally(() => {
+              if (typeof callback === 'function') {
+                callback();
+              }
+            });
+          },
+          reportCheckout(listUniqueId, listName, brandName, callback = null) {
+            const date = new Date();
+            fetch(
+              '/api/analytics/checkout',
+              {
+                method: 'POST',
+                credentials: 'include',
+                headers: {
+                  'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                  list_id: listUniqueId,
+                  list_name: listName,
+                  item_brand: brandName,
+                  created_at: date.toISOString(),
+                })
+              }
+            )
+            .then(response => {
+              if (response.status === 200) {
+                return response.json();
+              }
+              return Promise.reject(response);
+            })
+            .catch((error) => {
+              console.error('reportCheckout error:', error);
+            })
+            .finally(() => {
+              if (typeof callback === 'function') {
+                callback();
+              }
+            });
+          },
+          getItemUniqueId(listID) {
+            for (let key in this.analyticsProductInfo.itemsMapping) {
+              if (listID.includes(key)) {
+                return this.analyticsProductInfo.itemsMappingitemsMapping[key];
+              }
+            }
+            return null;
+          },
           addPopupGtagEvent() {
-            gtag("event", "view_item", {
-              items: [
+            const listUniqueId = this.getItemUniqueId(this.popup.URL) || this.popup.URL;
```

Comment:
this code block to get `listUniqueId`, `domain` and `items` seems appears twice, should it be a function?